### PR TITLE
Disable click outside dialog to dismiss for connect flows

### PR DESCRIPTION
**Pull request by microbit-grace**
https://microbit-global.monday.com/boards/1356069004/pulses/1356475624

Preview: https://dismiss-outside-dialog.ml-trainer.pages.dev/

### CODE REVIEW

File: src/components/dialogs/StandardDialog.svelte [hunk 1]
```diff
@@ -36,7 +37,10 @@
 <div class="relative z-10">
   <Transition show={$dialog.expanded}>
     <div
-      class="fixed top-0 left-0 h-screen w-screen flex justify-center items-center bg-black/50 bg-blend-darken">
+      class="fixed top-0 left-0 h-screen w-screen flex justify-center items-center bg-black/50 bg-blend-darken"
+      use:dialog.modal
+      role="none"
```
Comment (microbit-robert) [inline src/components/dialogs/StandardDialog.svelte:42]: I know it complains, but this element dynamically gets the correct role of "dialog", I don't think we should be overwriting that to "none" here.

**Review by microbit-robert — CHANGES_REQUESTED** [review]
Clicking inside a dialog that has dismissOnClickOutside set to true causes it to close due to event propagation which needs to be addressed.

**Comment by microbit-robert** [comment]
This third-party component makes me sad. Once you click inside the dialog, focus gets set to the body element and the escape key no longer closes the dialog. Not something you've changed, just an existing bug.

**Comment by microbit-grace** [comment]
> Clicking inside a dialog that has dismissOnClickOutside set to true causes it to close due to event propagation which needs to be addressed.

Nice catch, I have prevented the propagation and removed the `role` here: https://github.com/microbit-foundation/ml-trainer/pull/117/commits/c47aba33fccae6f218704295d57d083daa9bce09


> This third-party component makes me sad. Once you click inside the dialog, focus gets set to the body element and the escape key no longer closes the dialog. Not something you've changed, just an existing bug.

😮 Oh no, I didn't notice that. That's annoying

**Review by microbit-robert — APPROVED** [review]
LGTM. This works well, but is definitely a work around. In an ideal world we'd have a modal component that allows more customisation.

**Pull request merged**